openstack/sharedfilesystems/v2/shares/results.go

```diff
@@ -324,3 +324,65 @@ type ExtendResult struct {
 type ShrinkResult struct {
 	gophercloud.ErrResult
 }
+
+// GetMetadataResult contains the response body and error from a GetMetadata request.
+type GetMetadataResult struct {
+	gophercloud.Result
+}
+
+// Extract will get the string-string map from GetMetadataResult
+func (r GetMetadataResult) Extract() (map[string]string, error) {
+	var s struct {
+		Metadata map[string]string `json:"metadata"`
+	}
+	err := r.ExtractInto(&s)
+	return s.Metadata, err
+}
+
+// GetMetadatumResult contains the response body and error from a GetMetadatum request.
+type GetMetadatumResult struct {
+	gophercloud.Result
+	key string
```

Comment:
I understand the purpose of this, but let's use the style that has been previously used here: https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/results.go#L346-L352

This returns a `map[string]string` so the user/dev can do:

```go
result, err := shares.GetMetadatum(c, shareID, "foo").Extract()
fmt.Println(result["foo"])
```

Once it's updated like this, it looks like GetMetadata, GetMetadatum, SetMetadata, and UpdateMetadata can all share a single result type:

```go
type MetadataResult struct {
  gophercloud.Result
}

func (r MetadataResult) Extract() (map[string]string, error) {
 ...
}
```

Let me know if this makes sense or if you have any questions.

Reply:
Actually `GetMetadatum` would still need a separate result type because it uses `meta` parent in its JSON response instead of `metadata` like the other methods use (see example responses for [show-metadata](https://github.com/openstack/manila/blob/master/api-ref/source/samples/share-show-metadata-response.json) VS [show-metadata-item](https://github.com/openstack/manila/blob/master/api-ref/source/samples/share-show-metadata-item-response.json)). I'll merge the result types for the rest of the methods.

Reply:
Should `GetMetadatumResult` then still return `map[string]string` like you're suggesting, or rather keep things as they are in this PR and return `string`? I'm totally fine with either, especially if it's to keep things consistent...although the latter seems a bit more user-friendly by not having to repeat the metadata key.

Reply:
> Actually GetMetadatum would still need a separate result type because it uses meta parent in its JSON response instead of metadata like the other methods use 

Good catch. 

> Should GetMetadatumResult then still return map[string]string like you're suggesting,

Yes, please.

I agree that it can be more intuitive to have only the value returned, but using this method keeps things consistent with other metadata areas in Gophercloud. In addition, it more closely models the API response (since the API response is a dict/object/hash).